redis.ts

```diff
@@ -2359,11 +2359,31 @@ class RedisImpl implements Redis {
   }
 }
 
-export interface RedisConnectOptions extends RedisConnectionOptions {
+interface RedisConnectionOptionsTCP extends BaseRedisConnectionOptions {
+  tls?: boolean;
   hostname: string;
   port?: number | string;
 }
 
+interface RedisConnectionOptionsUnix extends BaseRedisConnectionOptions {
+  path: string;
+}
+
+// @ts-ignore: Exploiting missing type resolving to `any` to detect if `UnixConnectOptions` is in type scope:
+type ExtendsUCO<T> = T extends Deno.UnixConnectOptions ? true : false;
+type ValidUCO = ExtendsUCO<{ path: string; transport: "unix" }>;
+type InvalidUCO = ExtendsUCO<{ transport: "not-unix" }>;
+type HasUCO = ValidUCO extends false ? never
+  : InvalidUCO extends true ? never
+  : {};
+
+export type RedisConnectionOptions =
+  | RedisConnectionOptionsTCP
+  | (HasUCO & RedisConnectionOptionsUnix);
```

Comment:
I think this definition could be simpler.
```suggestion
  | RedisConnectionOptionsUnix;
```